src/core/reducers/addons.js

```diff
@@ -225,13 +224,6 @@ export function createInternalAddon(
     );
   }
 
-  // Remove undefined properties entirely. This is for some legacy code
-  // in Discopane that relies on spreads to combine a Discopane result
-  // (which has a header and description) with a minimal add-on object.
-  // For example, the minimal add-on object does not have a description
-  // property so the spread should not override `description`.
-  addon = removeUndefinedProps(addon);
```

Comment:
You're bold. I know it mentions discopane but I didn't remove it before because I was wondering if we didn't start to rely on that for other stuff later.

Reply:
I did a bunch of local testing, and everything seems okay. We should be able to handle undefined props.